core/src/main/java/org/apache/accumulo/core/spi/balancer/SimpleLoadBalancer.java

```diff
@@ -0,0 +1,380 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.spi.balancer;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.accumulo.core.client.AccumuloException;
+import org.apache.accumulo.core.client.AccumuloSecurityException;
+import org.apache.accumulo.core.data.TableId;
+import org.apache.accumulo.core.data.TabletId;
+import org.apache.accumulo.core.manager.balancer.TabletServerIdImpl;
+import org.apache.accumulo.core.spi.balancer.data.TServerStatus;
+import org.apache.accumulo.core.spi.balancer.data.TableStatistics;
+import org.apache.accumulo.core.spi.balancer.data.TabletMigration;
+import org.apache.accumulo.core.spi.balancer.data.TabletServerId;
+import org.apache.accumulo.core.spi.balancer.data.TabletStatistics;
+import org.apache.accumulo.core.spi.balancer.util.ThrottledBalancerProblemReporter;
+import org.apache.accumulo.core.spi.balancer.util.ThrottledBalancerProblemReporter.OutstandingMigrationsProblem;
+import org.apache.accumulo.core.spi.balancer.util.ThrottledBalancerProblemReporter.Problem;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A simple tablet balancer that attempts to spread tablets out evenly across all available tablet
+ * servers. The goal is to achieve the same number of tablets on each tablet server.
+ *
+ * <p>
+ * Initial assignments attempt to maintain locality by assigning tablets to their previous location
+ * if possible, but otherwise assignments are made in a random fashion across all available tablet
+ * servers.
+ *
+ * @since 2.1.0
+ */
+public class SimpleLoadBalancer implements TabletBalancer {
```

Comment:
I assume this was just copied from `DefaultLoadBalancer`? It would be nice to mention this in the javadoc since it is difficult to track a file that was moved and renamed. The

Reply:
Would it be better to have a reference in the docs for the deprecated DefaultLoadBalancer so that someone who is looking to upgrade will know that this is the replacement? It looks like the deprecation on there points to the TabletBalancer interface.

Reply:
On #1905 I made the old plugin impls log a warning mentioning the new impl class to configure instead. Maybe could do that here and if someone configures the  deprecated DefaultLoadBalancer it will log a warning encouraging use of the SimpleLoadBalancer (fully qualified) instead.

Reply:
Just some reference in both is fine. I was just thinking of a developer who has to make a change to one and needs to find the other one to either backport it or migrate it forward.

Reply:
I added warnings to constructors for the deprecated balancers, and updated javadoc to DefaultLoadBalancer and SimpleLoadBalancer referencing each other.